nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -102,16 +102,37 @@ fi
 extraBuildFlags+=(--option "build-users-group" "$buildUsersGroup")
 
 # Inherit binary caches from the host
-# TODO: will this still work with Nix 1.12 now that it has no perl? Probably not... 
+# TODO: will this still work with Nix 1.12 now that it has no perl? Probably not...
 binary_caches="$(@perl@/bin/perl -I @nix@/lib/perl5/site_perl/*/* -e 'use Nix::Config; Nix::Config::readConfig; print $Nix::Config::config{"binary-caches"};')"
 extraBuildFlags+=(--option "binary-caches" "$binary_caches")
 
 nixpkgs="$(readlink -f "$(nix-instantiate --find-file nixpkgs)")"
 export NIX_PATH="nixpkgs=$nixpkgs:nixos-config=$mountPoint/$NIXOS_CONFIG"
 unset NIXOS_CONFIG
 
-# TODO: do I need to set NIX_SUBSTITUTERS here or is the --option binary-caches above enough?
-
+# Set up some bind mounts we'll want regardless of chroot or not
+mount --rbind /dev "$mountPoint/dev"
+mount --rbind /proc "$mountPoint/proc"
+mount --rbind /sys "$mountPoint/sys"
```

Comment:
In case of doing a fresh install those mount points don't exist yet.

Reply:
We only support doing this from some sort of live cd installation, as far as I know, which will have those mount points. I'm just replicating the old installer's behavior here, not trying to add new functionality.

Reply:
AFAICT those don't exist until `nixos-prepare-root` is ran later on in the installer, and installation starts of with more or less empty `$mountPoint`:

```
$ nix-build "nixos/release-combined.nix" -A nixos.tests.installer.simple.x86_64-linux
# bunch of noise omitted
machine: must succeed: nixos-install < /dev/null >&2
machine# > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > mount: /mnt/dev: mount point does not exist.
```

Reply:
@pbogdan that's why I consider this option to be pretty awkward, but my understanding is that you're only expected to run `--chroot` on an "installed" (i.e., you've already run install on it once at some point in the past) `$mountPoint`. It really doesn't feel to me like this functionality belongs in the installer itself, but rather in an adjoining tool that's called something like "poke-around-post-install.sh".

So at that point, `$mountPoint` has the folders. I think?

Reply:
Yes, for `--chroot` the assumption is that you have an existing install somewhere and all of those are present. I'm only trying to point to out that as is this change would break the use case of using the installer as, well, installer 😆.
And I agree that it seems like somewhat strange functionality to be included, and recall initially being a bit confused as to what purpose it was supposed to serve in context of an installer.

Reply:
Oh, good point, sorry for being dense 😄  I'll push up a fix later